substrate/client/transaction-pool/src/builder.rs

```diff
@@ -0,0 +1,218 @@
+					count: 100_000,
+					total_bytes: 100 * 1024 * 1024,
+				},
+				future: crate::graph::base_pool::Limit {
+					count: 100_000,
+					total_bytes: 100 * 1024 * 1024,
+				},
+				reject_future_transactions: false,
+				ban_time: Duration::from_secs(30 * 60),
+			},
+			txpool_type: TransactionPoolType::SingleState,
+		}
+	}
+}
+
+use crate::{common::api::FullChainApi, graph::ChainApi};
+
+/// `FullClientTransactionPool` is a trait that combines the functionality of
+/// `MaintainedTransactionPool` and `LocalTransactionPool` for a given `Client` and `Block`.
+///
+/// This trait defines the requirements for a full client transaction pool, ensuring
+/// that it can handle transactions submission and maintenance.
+pub trait FullClientTransactionPool<Block, Client>:
+	MaintainedTransactionPool<
+		Block = Block,
+		Hash = crate::graph::ExtrinsicHash<FullChainApi<Client, Block>>,
+		InPoolTransaction = crate::graph::base_pool::Transaction<
+			crate::graph::ExtrinsicHash<FullChainApi<Client, Block>>,
+			<Block as BlockT>::Extrinsic,
+		>,
+		Error = <FullChainApi<Client, Block> as ChainApi>::Error,
+	> + LocalTransactionPool<
+		Block = Block,
+		Hash = crate::graph::ExtrinsicHash<FullChainApi<Client, Block>>,
+		Error = <FullChainApi<Client, Block> as ChainApi>::Error,
+	>
+where
+	Block: BlockT,
+	Client: sp_api::ProvideRuntimeApi<Block>
+		+ sc_client_api::BlockBackend<Block>
+		+ sc_client_api::blockchain::HeaderBackend<Block>
+		+ sp_runtime::traits::BlockIdTo<Block>
+		+ sp_blockchain::HeaderMetadata<Block, Error = sp_blockchain::Error>
+		+ 'static,
+	Client::Api: sp_transaction_pool::runtime_api::TaggedTransactionQueue<Block>,
+{
+}
+
+impl<Block, Client, P> FullClientTransactionPool<Block, Client> for P
+where
+	Block: BlockT,
+	Client: sp_api::ProvideRuntimeApi<Block>
+		+ sc_client_api::BlockBackend<Block>
+		+ sc_client_api::blockchain::HeaderBackend<Block>
+		+ sp_runtime::traits::BlockIdTo<Block>
+		+ sp_blockchain::HeaderMetadata<Block, Error = sp_blockchain::Error>
+		+ 'static,
+	Client::Api: sp_transaction_pool::runtime_api::TaggedTransactionQueue<Block>,
+	P: MaintainedTransactionPool<
+			Block = Block,
+			Hash = crate::graph::ExtrinsicHash<FullChainApi<Client, Block>>,
+			InPoolTransaction = crate::graph::base_pool::Transaction<
+				crate::graph::ExtrinsicHash<FullChainApi<Client, Block>>,
+				<Block as BlockT>::Extrinsic,
+			>,
+			Error = <FullChainApi<Client, Block> as ChainApi>::Error,
+		> + LocalTransactionPool<
+			Block = Block,
+			Hash = crate::graph::ExtrinsicHash<FullChainApi<Client, Block>>,
+			Error = <FullChainApi<Client, Block> as ChainApi>::Error,
+		>,
+{
+}
+
+/// The type alias for a dynamic trait object implementing
+/// `FullClientTransactionPool` with the given `Client` and `Block` types.
+///
+/// This trait object abstracts away the specific implementations of the transaction pool.
+pub type TransactionPoolImpl<Block, Client> = dyn FullClientTransactionPool<Block, Client>;
```

Comment:
Nit. I'm not sure about this name, I'm referring to the `Impl` suffix. But I can't suggest a better name :-)